src/Model/Todo/Handler/AddDeadlineToTodoHandler.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Prooph\ProophessorDo\Model\Todo\Handler;
+
+use Prooph\ProophessorDo\Model\Todo\Command\AddDeadlineToTodo;
+use Prooph\ProophessorDo\Model\Todo\TodoList;
+
+/**
+ * Class AddDeadlineToTodoHandler
+ *
+ * @package Prooph\ProophessorDo\Model\Todo\Handler
+ * @author Wojtek Gancarczyk <wojtek@aferalabs.com>
+ */
+final class AddDeadlineToTodoHandler
+{
+    /**
+     * @var TodoList
+     */
+    private $todoList;
+
+    /**
+     * @param TodoList $todoList
+     */
+    public function __construct(TodoList $todoList)
+    {
+        $this->todoList = $todoList;
+    }
+
+    /**
+     * @param AddDeadlineToTodo $command
+     * @return void
+     * @throws \Exception
+     */
+    public function __invoke(AddDeadlineToTodo $command)
+    {
+        $todo = $this->todoList->get($command->todoId());
+
+        if ($todo->assigneeId()->toString() !== $command->userId()->toString()) {
```

Comment:
Move this check into `Todo::addDeadline` method by passing the userId from the command to the Todo. 
Background: The aggregate is responsible for protecting invariants. The handler only locates the correct aggregate and triggers the action.

Also a dedicated exception class would be nice. See [existing exceptions](https://github.com/prooph/proophessor-do/tree/master/src/Model/Todo/Exception) for inspiration.